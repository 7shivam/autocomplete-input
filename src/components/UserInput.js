import React from 'react';

const UserInput = ({
    setUser,
    value,
    onKeyDown
}) => {
    return (
        <div className="input-container">
            <input
                placeholder="Search users"
                className="mt-5 input"
                type="text"
                value={value}
                onChange={(e) => setUser(e.target.value)}
                onKeyDown={onKeyDown}
            />
        </div>
    )
}

export default UserInput;
