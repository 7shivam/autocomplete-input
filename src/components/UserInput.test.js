import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import UserInput from './UserInput';

test('renders user input', () => {
    const tree = renderer
        .create(
            <UserInput
                value="Ross"
                onKeyDown={() => {}}
                setUser={() => {}}
            />
        ).toJSON();
    
    expect(tree).toMatchSnapshot();
})

test('call setUser when user type in the input', () => {
    const setUser = jest.fn();

    render(
        <UserInput
            value=""
            onKeyDown={() => {}}
            setUser={setUser}
        />
    )

    userEvent.type(screen.getByPlaceholderText('Search users'), 'M')
    expect(setUser).toHaveBeenCalledWith('M');
})

test('call onKeyDown when user press navigation key', () => {
    const onKeyDown = jest.fn();

    render(
        <UserInput
            value=""
            onKeyDown={onKeyDown}
            setUser={() => {}}
        />
    )

    fireEvent.keyDown(screen.getByPlaceholderText('Search users'), {
        keyCode: 38
    })
    expect(onKeyDown).toHaveBeenCalled();
})

test('call onKeyDown when user press enter', () => {
    const onKeyDown = jest.fn();

    render(
        <UserInput
            value=""
            onKeyDown={onKeyDown}
            setUser={() => {}}
        />
    )

    fireEvent.keyDown(screen.getByPlaceholderText('Search users'), {
        keyCode: 13
    })
    expect(onKeyDown).toHaveBeenCalled();
})
