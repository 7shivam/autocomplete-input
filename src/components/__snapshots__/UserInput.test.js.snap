// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user input 1`] = `
<div
  className="input-container"
>
  <input
    className="mt-5 input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search users"
    type="text"
    value="Ross"
  />
</div>
`;
